### PR TITLE
Fix #17: Use Extension.get_data_dir() as default image directory.

### LINKED ISSUE #17: Use Extension().get_data_dir()
See mopidy/mopidy-local-sqlite#66. Note that if we depend on Mopidy >1.1, `Extension.get_data_dir()` will be available, so no need to check...

### PR DESCRIPTION
**Pull request by tkem**
Basically follows https://github.com/mopidy/mopidy-local-sqlite/pull/66, but with a slightly different (less ugly?) workaround to make `Extension.get_data_dir()` available from a `local.Library`.
Also removes some pre-Mopidy v1.1 cruft.

### CODE REVIEW

**Comment by tkem** [comment]
@jodal: Any ideas about Travis `ImportError: No module named gobject`?
Never had this before moving to mopidy org...

**Comment by jodal** [comment]
There was a launchpad/PPA network glitch, it seems. I just reran the tests.

File: mopidy_local_images/__init__.py [hunk 1]
```diff
@@ -31,28 +30,33 @@ def get_config_schema(self):
...
             (r'/(.+)', ImageHandler, {'path': image_dir})
         ]
+
+    def _get_or_migrate_data_dir(self, config):
+        # TODO: Remove this method when we're confident most users
+        # have upgraded away from Mopidy 1.0.
+        new_dir = self.get_data_dir(config)
+        old_dir = os.path.join(config['core']['data_dir'], b'local', b'images')
+        if os.path.isdir(old_dir):
+            logger.info('Migrating Mopidy-Local-SQLite to new data dir')
```
Comment (jodal) [inline mopidy_local_images/__init__.py]: Wrong extension name!

File: mopidy_local_images/library.py [hunk 2]
```diff
@@ -228,3 +219,6 @@ def _scan(self, uri):
         logger.debug('Scanning %s for images', uri)
         data = self.scanner.scan(uri)
         return data.get('tags', {})
+
+    def get_data_dir(self, config):
+        raise NotImplementedError  # overridden by Extension
```
Comment (jodal) [inline mopidy_local_images/library.py]: I don't think there's anything the migrate function needs from the Extension. Might as well just move it here.

File: mopidy_local_images/__init__.py (hunk 1 above)
Comment (tkem) [inline mopidy_local_images/__init__.py]: Thanks; this much more fun since we migrated this to the mopidy org ;-)

**Comment by jodal** [comment]
Would be nice to have this split in two commits:

- Require Mopidy 1.1 and remove `ImportError` workarounds
- Switch to `Extension.get_data_dir()`

File: mopidy_local_images/library.py (hunk 2 above)
Comment (tkem) [inline mopidy_local_images/library.py]: Can you explain a bit more... You mean split the get_or_migrate_data_dir() into two functions?

Comment (jodal) [inline mopidy_local_images/library.py]: That's also a good idea.

Another thing: the function here should be private so it is not exposed to other Pykka actors.

``` python
    def _get_data_dir(self, config):
        self._migrate_old_data_dir(config)
        return Extension().get_data_dir(config)

    def _migrate_old_data_dir(self, config):
        # TODO: Remove this method when we're confident most users
        # have upgraded away from Mopidy 1.0.
        new_dir = Extension().get_data_dir(config)
        old_dir = os.path.join(config['core']['data_dir'], b'local', b'images')
        if os.path.isdir(old_dir):
            logger.info('Migrating Mopidy-Local-SQLite to new data dir')
            for filename in os.listdir(old_dir):
                old_path = os.path.join(old_dir, filename)
                new_path = os.path.join(new_dir, filename)
                logger.info('Moving %r to %r', old_path, new_path)
                os.rename(old_path, new_path)
            os.rmdir(old_dir)
```

**Comment by tkem** [comment]
- Split into two commits as suggested.
- get_data_dir/migrate stuff now copied from mopidy-local-sqlite (with s/sqlite/images); so if we want to change something, we can simply do it in both extensions

**Pull request merged**